plugins/snowflake/dbt/include/snowflake/macros/adapters.sql

```diff
@@ -52,31 +52,25 @@
 {% endmacro %}
 
 {% macro snowflake__get_columns_in_relation(relation) -%}
-  {% call statement('get_columns_in_relation', fetch_result=True) %}
-      select
-          column_name,
-          data_type,
-          character_maximum_length,
-          numeric_precision,
-          numeric_scale
-
-      from
-      {{ relation.information_schema('columns') }}
-
-      where table_name ilike '{{ relation.identifier }}'
-        {% if relation.schema %}
-        and table_schema ilike '{{ relation.schema }}'
-        {% endif %}
-        {% if relation.database %}
-        and table_catalog ilike '{{ relation.database }}'
-        {% endif %}
-      order by ordinal_position
+  {%- set sql -%}
+    describe table {{ relation }}
+  {%- endset -%}
+  {%- set result = run_query(sql) -%}
 
-  {% endcall %}
-
-  {% set table = load_result('get_columns_in_relation').table %}
-  {{ return(sql_convert_columns_in_relation(table)) }}
+  {% set maximum = 10000 %}
+  {% if (result | length) >= maximum %}
+    {% set msg %}
+      Too many columns in relation {{ relation }}! dbt can only get
+      information about relations with fewer than {{ maximum }} columns.
```

Comment:
i hope nobody ever sees this error message :p

Reply:
I actually don't know for sure that this is a limit, I just assumed it based on all the other commands. I'm comfortable saying that if someone has 10k columns in a table, we can revisit this then :)